File: src/AzureDataFactory.TestingFramework/Models/Activities/ControlActivity/ControlActivity.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

using AzureDataFactory.TestingFramework.Models.Activities.Base;
using AzureDataFactory.TestingFramework.Models.Pipelines;

namespace AzureDataFactory.TestingFramework.Models;

public partial class ControlActivity
{
    internal delegate IEnumerable<PipelineActivity> EvaluateActivitiesDelegate(List<PipelineActivity> activities, PipelineRunState state);
    internal virtual IEnumerable<PipelineActivity> EvaluateControlActivityIterations(PipelineRunState state,  EvaluateActivitiesDelegate evaluateActivities)
    {
        // Note: unfortunately cannot use abstract method
        return new List<PipelineActivity>();
    }
}

File: src/python/data_factory_testing_framework/models/activities/control_activities/control_activity.py
from typing import Callable, Generator

from data_factory_testing_framework.generated.models import Activity
from data_factory_testing_framework.models.state.pipeline_run_state import PipelineRunState


class ControlActivity:
    def evaluate_control_activity_iterations(
        self,
        state: PipelineRunState,
        evaluate_activities: Callable[[PipelineRunState], Generator[Activity, None, None]],
    ):
        return []
